### ISSUE TITLE
User Review

### ISSUE THREAD

**Issue by justineechen** [OpenLiberty/guide-sessions#40]
User Review:

Allow a person with a fresh mind to read the guide and execute the steps described in the guide, test whether each step is clear enough to understand, and no logic errors.

Review and walkthrough by a non-SME or a new user with a critical and fresh pair of eyes:

- [x] Record the time it takes to go through the entire guide
- [x] Test each step in the guide is clear and easy to understand
- [x] Ensure no logic errors
- [x] Resolve anything that does not work
- [x] Resolve anything usability issues (confusing or hard to follow)
- [x] Achieve basic understanding/working knowledge of the topic in the guide

**Comment by wojtczat** [comment]
## What you'll learn 
- [ ] "You'll also learn" For some reason on the test site there is a space between the apostrophe and "u" in "you'll". It looks like it is probably an encoding error. When comparing the apostrophe to other occurrences of "you'll" in the guide there is a noticeable difference between them. You'll notice if you paste the sentence into VS Code, the space will not be copied over; however, when you paste that line from the editor back into one a comment box like this for e.g. you'll notice the character before the apostrophe cannot be recognized.
- [ ] The following two paragraphs on application containerization and Kubernetes do not flow well. I am sure they are being presented because the two topics were presented in the prior paragraph however I find this sequencing poor.

## Getting started
- [ ] First line of git clone command overruns the line not sure if this was ever fixed.

 ## Creating the application
 - [ ] Kind of messy indentation on the return in the `getHostname()` method in the `CartResource` class.
 - [ ] "The [hotspot=endpointCart file=1] cart endpoint demonstrates how to get session data." probably need to fix this hotspot
 
## Configuring session persistence
- [ ] "instances through the <group/> configuration tag" should "<group/>" be presented/hotspotted in a similar way that the "library" tag above it is hotspotted?
- [ ] "hazelcast-config.xml" should probably have a hotspot if the server.xml is hotspotted.

## Deploying and running the containerized application
#### FYI 
If you are working in a custom namespace in ICP you will receive an error when running `kubectl apply -f kubernetes.yaml` as ICP controls where images are deployed from. So on an initial run you might see an error like 
```
Error from server (InternalError): error when creating "kubernetes.yaml": Internal error occurred: admission webhook "trust.hooks.securityenforcement.admission.cloud.ibm.com" denied the request:
Deny "docker.io/cart-app:1.0-SNAPSHOT", no matching repositories in ClusterImagePolicy and no ImagePolicies in the <your_namespace> namespace
```
To fix this error: 
* Go to the cluster dashboard > Manage > Resource Security > Resource Security > Image Policy > Create Custom Image Policy 
* Fill out the form i.e. name of policy, scope (cluster or namespace), Registry Policies
* For registry URL put the value in this case: `docker.io/cart-app`, VA Scan Policy: either option

#### FYI 2
If someone is using a cluster with nodes that correspond to different architectures it is likely that the `kubectl apply -f kubernetes.yaml` command will only spawn pods that receive `ImagePullBackOff` errors. This will happen as there is no `nodeSelector` in the `kubernetes.yaml`. This would be a user issue, but I figured I would document it here just in case the issue is raised in the future. To resolve the issue corresponding to this problem, the user would want to add a `nodeSelector` field to the `kubernetes.yaml` file to enforce that the pods will run on the node with the correct architecture. If issues persist then there are likely to be issues with the node tolerations.  For more info see https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/ and https://kubernetes.io/docs/concepts/configuration/assign-pod-node/

- [ ] "You’ll see the session-id should be same but the pod-name is different" should be "the same"
- [ ] The code command underneath "You may resume the pod by the following command:" flows outside of the code command window/box without any opportunity to see the enter line without copying and pasting the command.

## Tearing down the environment
- [ ] I noticed that there were some tests that came with the guide, but the guide doesn't have the user run them. I thought I would make a point to ensure that you did not
forget to add anything to the guide.
- [ ] I also assume more tags could be included than just "Java EE"

Ignoring the issues that came with using a private registry the guide did not take too long to go through. ~25 minutes seems like an appropriate time.

**Comment by justineechen** [comment]
- added Docker and Microprofile in tags 
- UI issues I will bring up to the design team but most cannot be addressed through this repo

**Comment by justineechen** [comment]
> The code command underneath "You may resume the pod by the following command:" flows outside of the code command window/box without any opportunity to see the enter line without copying and pasting the command.

- I don't seem to have this issue, is this what you saw?
<img width="654" alt="Screen Shot 2019-06-27 at 2 50 02 PM" src="https://user-images.githubusercontent.com/18709242/60292343-e39d0b00-98ea-11e9-82c8-0d622232a822.png">

>First line of git clone command overruns the line not sure if this was ever fixed.

- I believe the UI team is working on the overrunning line/horizontal scrolling in boxes

**Comment by wojtczat** [comment]
Almost everything looks fine. The overflow on that kubectl command occurred on chrome on a 15in Mac. It cuts off after the "l" in "default". I am still uncertain about the flow of those 3 paragraphs in the second point I made. I just don't know how important it really is to explain containerization and kubernetes at that specific moment. It seems like it can be difficult to present all of those concepts in a way that doesn't come off as "choppy". I wonder if it could work better later in the guide? It may be worthwhile getting a second opinion on that issue.
